library/swift/HeadersBuilder.swift

```diff
@@ -3,14 +3,17 @@ import Foundation
 private let kRestrictedPrefixes = [":", "x-envoy-mobile"]
 
 private func isRestrictedHeader(name: String) -> Bool {
-  return name == "host" || kRestrictedPrefixes.contains { name.hasPrefix($0) }
+  let lowercasedName = name.lowercased()
+  return lowercasedName == "host" || kRestrictedPrefixes.contains { lowercasedName.hasPrefix($0) }
```

Comment:
Eagerly converting each character to its lowercase form could be expensive.

Can you use Foundation's case-insensitive comparison APIs instead? E.g. https://developer.apple.com/documentation/foundation/nsstring/1416849-range

Reply:
good idea, Note - we will still need to lowercase inside of the implementation of `HeadersContainer`. We could probably avoid that if we decide to use some 'smart' container in there but not sure whether it's worth doing for now.

Reply:
We could do some tweaks to make Swift dictionary case-insensitive (i.e., https://stackoverflow.com/a/47134176) but not sure what's cheaper - iterating over all keys and doing case insensitive comparisons or lowercasing the header name. I could do some benchmarking here but my light preference (I am biased) it to keep the lowercasing operations they are now. Let me know what you think.